webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/score/ComplianceScore.scala

```diff
@@ -0,0 +1,84 @@
+/*
+ *************************************************************************************
+ * Copyright 2024 Normation SAS
+ *************************************************************************************
+ *
+ * This file is part of Rudder.
+ *
+ * Rudder is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * In accordance with the terms of section 7 (7. Additional Terms.) of
+ * the GNU General Public License version 3, the copyright holders add
+ * the following Additional permissions:
+ * Notwithstanding to the terms of section 5 (5. Conveying Modified Source
+ * Versions) and 6 (6. Conveying Non-Source Forms.) of the GNU General
+ * Public License version 3, when you create a Related Module, this
+ * Related Module is not considered as a part of the work and may be
+ * distributed under the license agreement of your choice.
+ * A "Related Module" means a set of sources files including their
+ * documentation that, without modification of the Source Code, enables
+ * supplementary functions or services in addition to those offered by
+ * the Software.
+ *
+ * Rudder is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Rudder.  If not, see <http://www.gnu.org/licenses/>.
+
+ *
+ *************************************************************************************
+ */
+
+
+package com.normation.rudder.score
+
+import com.normation.errors.PureResult
+import com.normation.inventory.domain.InventoryError.Inconsistency
+import com.normation.inventory.domain.NodeId
+import com.normation.rudder.domain.reports.ComplianceSerializable
+import com.normation.rudder.score.ScoreValue.A
+import com.normation.rudder.score.ScoreValue.B
+import com.normation.rudder.score.ScoreValue.C
+import com.normation.rudder.score.ScoreValue.D
+import com.normation.rudder.score.ScoreValue.E
+import zio.json.DeriveJsonEncoder
+import zio.json.EncoderOps
+import zio.json.JsonEncoder
+
+object ComplianceScoreEventHandler extends ScoreEventHandler {
+  implicit val compliancePercentEncoder: JsonEncoder[ComplianceSerializable]     = DeriveJsonEncoder.gen
+  def handle(event: ScoreEvent):         PureResult[List[(NodeId, List[Score])]] = {
```

Comment:
it looks like we always have that structure: 
- a `Score` computation from a subtype of `ScoreEvent`
- a partial function from a `ScoreEvent` to a `Score` (ie a "can handle that type" + "if yes, use the computation function with the correct subtype of score event"

I think you need to split appart these two bits so that the `score from score event computation` is unit-testable by itself on two aspects:
- the correctness of the computation (easy here, but it will be a kind of spec so users will question it a lot, we need to be able to accumulate all the strange corner case they will find)
- the JSON format, because it becomes a storage format that will need to never change without a migration

Plus, once you have that separation, the implementation of the `handle` will be almost automatic, so more easy for other people needing to do so to understand what's going on.